x-pack/filebeat/module/zeek/dns/config/dns.yml

```diff
@@ -6,39 +6,132 @@ paths:
 exclude_files: [".gz$"]
 tags: {{.tags}}
 
-json.keys_under_root: false
-
 processors:
-  - drop_fields:
-      fields: ["json.Z","json.auth","json.addl"]
   - rename:
       fields:
-        - from: "json"
-          to: "zeek.dns"
-
-        - from: "zeek.dns.id.orig_h"
-          to: "source.address"
+        - {from: message, to: log.original}
+  - decode_json_fields:
+      fields: [log.original]
+      target: zeek.dns
+  - script:
+      lang: javascript
+      id: zeek_dns_flags
+      source: >
+        function addDnsHeaderFlags(evt) {
+            var flag = evt.Get("zeek.dns.AA");
+            if (flag === true) {
+                evt.AppendTo("dns.header_flags", "AA");
+            }
+            flag = evt.Get("zeek.dns.TC");
+            if (flag === true) {
+                evt.AppendTo("dns.header_flags", "TC");
+            }
+            flag = evt.Get("zeek.dns.RD");
+            if (flag === true) {
+                evt.AppendTo("dns.header_flags", "RD");
+            }
+            flag = evt.Get("zeek.dns.RA");
+            if (flag === true) {
+                evt.AppendTo("dns.header_flags", "RA");
+            }
+            flag = evt.Get("zeek.dns.Z");
```

Comment:
Not a big deal, but what's `Z`? 🤔

Looked it up, and it's actually a count. According to [their doc](https://docs.zeek.org/en/stable/scripts/base/protocols/dns/main.bro.html#type-DNS::Info), it's usually zero, and should be ignored.

I'd say ignore it, or at least don't add a `Z` flag to the DNS headers when it's there :-)